### PR TITLE
Polymer2.0 upgrade

### PR DESCRIPTION
**Pull request by davidrleonard**
* Upgrades component to be Polymer 1.x/2.x hybrid compatible
* Adds new property selectedItemRef that notifies out the selected item object.
  This makes it more convenient to bind the selected item and get information
  about it rather than looping over the listItems to find the one that matches
  the ID in selectedItem.
* Refactor to remove moment.js dependency in favor of a small fromNow method,
  adds localization strings to match. Reduces 16kb dependency that is only used for
  formatting time in "X time ago" strings and replace with 23 lines of code that is
  tested to conform to `moment.fromNow()` API.

@mdwragg @randyaskin could someone sign off on this, quite a few changes here but respects current API so we can just do a minor bump.

### CODE REVIEW

File: px-inbox.html [hunk 1]
```diff
@@ -187,129 +203,158 @@
...
             subtitle.indexOf(query) != -1);
         };
       }
     },
+    /**
+     * Reverses the sort order for the component.
+     */
+    _handleReverseSort: function() {
+      if (this.sortOrder === 'ascending') {
+        this.sortOrder = 'descending';
```
Comment (benoitjchevalier) [inline px-inbox.html:343]: If we make sortOrder notify, we probably should use set here

**Review by benoitjchevalier — COMMENTED** [review]
1 minor comment, otherwise LGTM

File: sass/px-inbox.scss [hunk 2]
```diff
@@ -120,13 +124,19 @@ px-dropdown {
   height: 100%;
   overflow-y: auto;
   flex: 1 1 auto;
-  &.list-active {
-    display: none;
-  }
+  // &.list-active {
```
Comment (mdwragg) [inline sass/px-inbox.scss:127]: Jus' delete any dead sass. Keep it clean etc. 👍

**Comment by mdwragg** [comment]
LGTM - just the comment above ^^ no biggy. 💯

**Review by randyaskin — APPROVED** [review]

**Pull request merged**